internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -60,20 +68,36 @@ func (c *CodePipeline) GetPipeline(pipelineName string) (*Pipeline, error) {
 	return pipeline, nil
 }
 
-// ListPipelines retrieves summaries of all pipelines for a project
-func (c *CodePipeline) ListPipelines() ([]string, error) {
-	input := &cp.ListPipelinesInput{}
-	resp, err := c.client.ListPipelines(input)
+// ListPipelines retrieves the names of all pipelines for a project
+func (c *CodePipeline) ListPipelinesForProject(projectName string) ([]string, error) {
+	var pipelineNames []string
+
+	tags := map[string]string{
+		"ecs-project": projectName,
+	}
+
+	arns, err := c.rgClient.GetResourcesByTags(pipelineResourceType, tags)
 	if err != nil {
-		return nil, fmt.Errorf("list pipelines: %w", err)
+		return nil, err
+	}
+
+	for _, arn := range arns {
+		name, err := c.getPipelineName(arn)
+		if err != nil {
+			return nil, err
+		}
+		pipelineNames = append(pipelineNames, name)
 	}
 
-	var pipelines []string
+	return pipelineNames, nil
+}
 
-	for _, ps := range resp.Pipelines {
-		p := aws.StringValue(ps.Name)
-		pipelines = append(pipelines, p)
+func (c *CodePipeline) getPipelineName(arn string) (string, error) {
+	i := strings.LastIndex(arn, ":")
+	if i == -1 {
+		return "", fmt.Errorf("cannot parse pipeline ARN: %s", arn)
```

Comment:
It might be safer to use the `arn` library provided by the SDK: https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/

Reply:
Yeah, I saw that was done in the cloudwatch lib but it seemed kind of heavyweight which is why I went with just using the strings stdlib